src/gar.lua

```diff
@@ -981,4 +982,21 @@ function gar.instantOperatorWithdrawal(from, vaultId, currentTimestamp)
 	}
 end
 
+function gar.isEligibleForArNSDiscount(from)
+	local gateway = gar.getGateway(from)
+	if gateway == nil then
+		return false
+	end
+
+	if gateway.weights == nil then
+		return false
+	end
+
+	local tenureWeight = gateway.weights.tenureWeight
+	local gatewayPerformanceRatio = gateway.weights.gatewayRewardRatioWeight
```

Comment:
nit: this shouldn't happen, but lua throws up when comparing numbers to `nil` so I would add `or 0` to both of these